File: 28. Find the Index of the First Occurrence in a String/Find the Index of the First Occurrence in a String.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
    let index = -1;
    const lenCheck = needle.length;
    for (let i = 0; i < haystack.length; i++) {
        if (i + lenCheck > haystack.length) return index;
        if (haystack.substring(i, i + lenCheck) == needle) {
            index = i;
            return index;
        }
        else {
            continue;
        }
    }
    return index;
};

File: 28. Find the Index of the First Occurrence in a String/Find the Index of the First Occurrence in a String.java

class Solution {

    public int strStr(String haystack, String needle) {
        int index = -1;
        int checkLength = needle.length();
        for (int i = 0; i < haystack.length(); i++) {
            if (i + checkLength > haystack.length()) return index;
            if (haystack.substring(i, i + checkLength).equals(needle)) {
                index = i;
                return index;
            }
            else {
                continue;
            }
        }
        return index;
    }
}
